demo/src/app/components/components.e2e-spec.ts

```diff
@@ -0,0 +1,52 @@
+import { browser, $ } from 'protractor';
+import { getLinkElement, getCodeToggleElement, getCodeElements } from '../tools.po';
+
+
+
+describe(`Components`, () => {
+
+  beforeAll(async() => {
+    await getLinkElement(`components`).click();
+  });
+
+  const components = [
+    'accordion', 'alert', 'buttons', 'carousel', 'collapse', 'datepicker', 'dropdown',
+    'modal', 'pagination', 'popover', 'progressbar', 'rating', 'table',
+    'tabset', 'timepicker', 'toast', 'tooltip', 'typeahead'
+  ];
+  components.forEach((component) => {
+    describe(`${component} page`, () => {
+
+
+      beforeAll(async() => {
+        await getLinkElement(`components/${component}`).click();
+      });
+
+      it (`should be displayed without errors`, async () => {
+        const browserLog = await browser.manage().logs().get('browser');
+        if (browserLog.length > 0) {
+          console.error(browserLog);
+          fail(`Unexpected console messages found`);
+        }
+      });
+
+      it(`should display the mains tabs`, async () => {
```

Comment:
Should be better if we
- get all the tabs from DOM of a component page, ex: `Overview`, `Examples`, `API`
- loop over them and click to open each tab